capfile/src/main/java/pro/javacard/sdk/SDKCompatibilityTable.java

```diff
@@ -0,0 +1,210 @@
+
+        // JavaCard 3.1.0 SDK version - supports targeting older versions
+        COMPATIBILITY_TABLE.put(SDKVersion.V310, new CompatibilityEntry(
+            SDKVersion.V310, "1.7", 1, 17, true, 
+            SDKVersion.V304, SDKVersion.V305, SDKVersion.V310, SDKVersion.V320));
+
+        // JavaCard 3.2.0 SDK versions - supports targeting older versions
+        COMPATIBILITY_TABLE.put(SDKVersion.V320, new CompatibilityEntry(
+            SDKVersion.V320, "1.7", 8, 17, true,
+            SDKVersion.V304, SDKVersion.V305, SDKVersion.V310, SDKVersion.V320));
+        
+        COMPATIBILITY_TABLE.put(SDKVersion.V320_24_1, new CompatibilityEntry(
+            SDKVersion.V320_24_1, "1.7", 11, 17, true,
+            SDKVersion.V304, SDKVersion.V305, SDKVersion.V310, SDKVersion.V320));
+        
+        COMPATIBILITY_TABLE.put(SDKVersion.V320_25_0, new CompatibilityEntry(
+            SDKVersion.V320_25_0, "1.8", 8, Integer.MAX_VALUE, true,
+            SDKVersion.V304, SDKVersion.V305, SDKVersion.V310, SDKVersion.V320));
+    }
+
+    /**
+     * Get compatibility information for an SDK version
+     */
+    public static CompatibilityEntry getCompatibility(SDKVersion version) {
+        CompatibilityEntry entry = COMPATIBILITY_TABLE.get(version);
+        if (entry == null) {
+            throw new IllegalArgumentException("Unknown SDK version: " + version);
+        }
+        return entry;
+    }
+
+    /**
+     * Get the Java class file version supported by an SDK
+     */
+    public static String getJavaClassFileVersion(SDKVersion version) {
+        return getCompatibility(version).getJavaClassFileVersion();
+    }
+
+    /**
+     * Check if a JDK version is compatible with an SDK version
+     */
+    public static boolean isJDKCompatible(SDKVersion sdkVersion, int jdkVersion) {
+        return getCompatibility(sdkVersion).isJDKCompatible(jdkVersion);
+    }
+
+    /**
+     * Check if an SDK version can target another SDK version for backwards compatibility
+     */
+    public static boolean canTarget(SDKVersion sourceSDK, SDKVersion targetSDK) {
+        return getCompatibility(sourceSDK).canTargetVersion(targetSDK);
+    }
+
+    /**
+     * Check if an SDK version supports multi-targeting
+     */
+    public static boolean isMultitarget(SDKVersion version) {
+        return getCompatibility(version).isMultitarget();
+    }
+
+    /**
+     * Get the minimum JDK version required for an SDK
+     */
+    public static int getMinJDKVersion(SDKVersion version) {
+        return getCompatibility(version).getMinJDK();
+    }
+
+    /**
+     * Get the maximum JDK version supported by an SDK
+     */
+    public static int getMaxJDKVersion(SDKVersion version) {
+        return getCompatibility(version).getMaxJDK();
+    }
+
+    /**
+     * Get all SDK versions that can be targeted by a given SDK version
+     */
+    public static Set<SDKVersion> getTargetableVersions(SDKVersion version) {
```

Comment:
write less code and single use setters. Don't create code comments that state the obvious. Use shorter names for methods.